Source/Core/InputCommon/ControlReference/ControlReference.cpp

```diff
@@ -4,19 +4,59 @@
 
 #include "InputCommon/ControlReference/ControlReference.h"
 
-using namespace ciface::ExpressionParser;
+#include "Core/Host.h"
 
-static thread_local bool tls_input_gate = true;
+using namespace ciface::ExpressionParser;
 
-void ControlReference::SetInputGate(bool enable)
+// See Control::FocusFlags for more details
+enum class InputGateFlags : u8
 {
-  tls_input_gate = enable;
-}
+  // This is always true if the user accepts background input, otherwise
+  // it's only true when the window has focus, similarly to the state below
+  HasFocus = 0x01,
+  // Full focus means the cursor is captured/locked inside the game window.
+  // Ignored if mouse capturing is off. Implies HasFocus
+  HasFullFocus = 0x02,
```

Comment:
Just going off the comment, maybe better as `RestrictMouseToWindow` ?

Reply:
It's not exactly that, when you don't have FullFocus other inputs can be ignored if you want, not only the mouse.
But yes, mainly, it's for the mouse. I know, it's an ambiguous name for an abstract concept, but I think the name explains it fine.